x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -305,15 +335,118 @@ func (k Keeper) proofRequirementForClaim(ctx sdk.Context, claim *prooftypes.Clai
+		claimeduPOKT,
 		proofParams.GetProofRequirementThreshold(),
 		proofRequirementSampleValue,
 		proofParams.GetProofRequestProbability(),
 	))
 	return requirementReason, nil
 }
 
+// slashSupplierStake slashes the stake of a supplier and mints the total slashing
+// amount to the tokenomics module account.
+func (k Keeper) slashSupplierStake(
+	ctx sdk.Context,
+	supplierOperatorAddress string,
+	slashingCount uint64,
+) error {
+	logger := k.logger.With("method", "slashSupplierStake")
+
+	proofParams := k.proofKeeper.GetParams(ctx)
+	slashingPenaltyPerExpiredClaim := proofParams.GetProofMissingPenalty()
+
+	totalSlashingAmt := slashingPenaltyPerExpiredClaim.Amount.Mul(math.NewIntFromUint64(slashingCount))
+	totalSlashingCoin := sdk.NewCoin(volatile.DenomuPOKT, totalSlashingAmt)
+
+	supplierToSlash, supplierFound := k.supplierKeeper.GetSupplier(ctx, supplierOperatorAddress)
+	if !supplierFound {
+		return tokenomicstypes.ErrTokenomicsSupplierNotFound
+	}
+
+	slashedSupplierInitialStakeCoin := supplierToSlash.GetStake()
+
+	var remainingStakeCoin sdk.Coin
+	if slashedSupplierInitialStakeCoin.IsGTE(totalSlashingCoin) {
+		remainingStakeCoin = slashedSupplierInitialStakeCoin.Sub(totalSlashingCoin)
+	} else {
+		logger.Warn(fmt.Sprintf(
+			"total slashing amount (%s) is greater than supplier %q stake (%s)",
+			totalSlashingCoin,
+			supplierOperatorAddress,
+			supplierToSlash.GetStake(),
+		))
+
+		// Set the remaining stake to 0 if the slashing amount is greater than the stake.
+		remainingStakeCoin = sdk.NewCoin(volatile.DenomuPOKT, math.NewInt(0))
+		// Total slashing amount is the whole supplier's stake.
+		totalSlashingCoin = sdk.NewCoin(volatile.DenomuPOKT, slashedSupplierInitialStakeCoin.Amount)
+	}
+
+	// Since staking mints tokens to the supplier module account, to have a correct
+	// accounting, the slashing amount needs to be sent from the supplier module
+	// account to the tokenomics module account.
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, suppliertypes.ModuleName, tokenomicstypes.ModuleName, sdk.NewCoins(totalSlashingCoin)); err != nil {
+		return err
+	}
+
+	supplierToSlash.Stake = &remainingStakeCoin
+
+	logger.Warn(fmt.Sprintf(
+		"slashing supplier owner with address %q operated by %q by %s, remaining stake: %s",
+		supplierToSlash.GetOwnerAddress(),
+		supplierToSlash.GetOperatorAddress(),
+		totalSlashingCoin,
+		supplierToSlash.GetStake(),
+	))
+
+	// Check if the supplier's stake is below the minimum and unstake it if necessary.
+	// TODO_BETA(@bryanchriswhite, #612): Use minimum stake governance parameter once available.
+	minSupplierStakeCoin := sdk.NewCoin(volatile.DenomuPOKT, math.NewInt(1))
+	// TODO_MAINNET(@red-0ne): Since SettlePendingClaims is not necessarily called
+	// at session end height so the unstaked supplier may not be immediately removed.
+	// Ensure that Gateways and Applications do not interact with a supplier that
+	// is below the minimum stake.
```

Comment:
```suggestion
	// TODO_MAINNET(@red-0ne): SettlePendingClaims is called at the end of every block,
	// but not every block corresponds to the end of a session. This may lead to a situation
	// where a force unstaked supplier may still be able to interact with a Gateway or Application.
	// However, claims are only processed when sessions end.
	// INVESTIGATION: This requires an investigation if the race condition exists at all and fixed
	// only if it does.
```